app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/DisplayProfile.kt

```diff
@@ -231,6 +279,127 @@ fun ProfileHeader(
               }
             }
       }
+
+  if (blockReportModal) {
+    val configuration = LocalConfiguration.current
+    val screenWidth = configuration.screenWidthDp.dp
+    val maxWidth = screenWidth * MAX_DIALOG_WIDTH_FACTOR
+
+    Dialog(
+        onDismissRequest = {
+          blockReportModal = false
+          showReportOptions = false
+          showBlockConfirmation = false
+        }) {
+          Card(
+              modifier =
+                  Modifier.widthIn(max = maxWidth)
+                      .padding(PADDING_STANDARD)
+                      .testTag("alertDialogReportBlock")) {
+                Column(
+                    modifier = Modifier.padding(PADDING_LARGE),
+                    verticalArrangement = Arrangement.spacedBy(8.dp)) {
+                      Text(
+                          text = stringResource(R.string.block_report_modal_title),
+                          style = MaterialTheme.typography.titleLarge,
+                          fontSize = modalTitleSize,
+                          modifier = Modifier.padding(bottom = PADDING_SMALL))
+                      // Content
+                      if (showReportOptions) {
+                        Column {
+                          Text("Select a reason for reporting:")
+                          reportType.values().forEach { reportType ->
+                            TextButton(
+                                onClick = { selectedReportType = reportType },
+                                modifier = Modifier.fillMaxWidth()) {
+                                  Text(
+                                      reportType.displayName,
+                                      color =
+                                          if (selectedReportType == reportType) IceBreakrrBlue
+                                          else Color.Black)
+                                }
+                          }
+                        }
+                      } else if (showBlockConfirmation) {
+                        Text(stringResource(R.string.block_confirmation_message))
+                      }
+                      // Buttons
+                      if (showReportOptions) {
+                        Row(
+                            modifier = Modifier.fillMaxWidth(),
+                            horizontalArrangement = Arrangement.SpaceBetween,
+                            verticalAlignment = Alignment.CenterVertically) {
+                              TextButton(
+                                  onClick = {
+                                    blockReportModal = false
+                                    showReportOptions = false
+                                    selectedReportType = null
+                                  }) {
+                                    Text("Cancel")
```

Comment:
Nitpick: Could move string to Strings.xml by creating a general cancel string that could be used for all popups